.github/workflows/maven.yml

```diff
@@ -37,3 +37,18 @@ jobs:
           cache: maven
       - name: Build with Maven
         run: mvn clean install --file pom.xml
+
+  java-21:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: '21'
+          distribution: 'adopt'
+          cache: maven
+      - name: Build with Maven
+        run: mvn clean install --file pom.xml
```

Comment:
Might be better to have a matrix of Java versions, rather than duplicate the same code block.  See something like https://github.com/openmrs/openmrs-core/blob/master/.github/workflows/build.yaml#L28